src/Symfony/Component/Security/Core/Authorization/Voter/AuthenticatedVoter.php

```diff
@@ -46,6 +48,10 @@ public function vote(TokenInterface $token, mixed $subject, array $attributes):
             return VoterInterface::ACCESS_GRANTED;
         }
 
+        if ($token instanceof OfflineTokenInterface) {
```

Comment:
this should not be done here. It should only be done when we identify an attribute on which we would not abstain. Otherwise, this voter would prevent any offline check that would be supported by a listener with a lower priority.

This must go inside the loop, after the initial checking for supported attributes.

Reply:
@stof looking at your comment below, are you also suggesting to abstain rather than throw an invalid argument exception? 

Personally I feel like that could create confusion as to why access isn't being provided and not being able to see specifically why. The exception makes it more obvious that a mistake is being made, rather than the user not having permissions that you'd otherwise expect them to have.

For example, if some service function works fine during testing in a controller or some session based context, if you try to run that same function when there isn't a session, it's not clear as to what the issue is. If there's an exception, the developer would be able to more easily identify that they're misusing the system.

Thoughts?

Reply:
@stof revisiting this I see what you're saying now, I've made the change for this.